core/src/test/scala/org/apache/spark/scheduler/BarrierTaskContextSuite.scala

```diff
@@ -275,7 +275,8 @@ class BarrierTaskContextSuite extends SparkFunSuite with LocalSparkContext with
   }
 
   test("SPARK-31485: barrier stage should fail if only partial tasks are launched") {
-    initLocalClusterSparkContext(2)
+    val conf = new SparkConf().set(LOCALITY_WAIT_PROCESS.key, "10s")
```

Comment:
Now that we have waited until all the executors has been launched before we submit any jobs, thus upon the first time we try to offer the resources to the pending tasks, we should expect exactly one task get launched, this shouldn't change following different locality wait time. cc @Ngone51

Reply:
I can't remember how far we backported the fix from @Ngone51 , is it in branch 2.4?

Reply:
The new delay scheduling update didn't go to 3.0 and previous versions, so this is a separated issue.